tests/functional/test_install_wheel.py

```diff
@@ -534,3 +534,34 @@ def test_wheel_installs_ok_with_nested_dist_info(script):
     script.pip(
         "install", "--no-cache-dir", "--no-index", package
     )
+
+
+def test_wheel_installs_ok_with_badly_encoded_irrelevant_metadata(script):
+    package = create_basic_wheel_for_package(
+        script,
+        "simple",
+        "0.1.0",
+        extra_files={
+            "simple-0.1.0.dist-info/AUTHORS.txt": b"\xff"
+        },
+    )
+    script.pip(
+        "install", "--no-cache-dir", "--no-index", package
+    )
+
+
+def test_wheel_install_fails_with_badly_encoded_metadata(script):
+    package = create_basic_wheel_for_package(
+        script,
+        "simple",
+        "0.1.0",
+        extra_files={
+            "simple-0.1.0.dist-info/METADATA": b"\xff"
+        },
+    )
+    result = script.pip(
+        "install", "--no-cache-dir", "--no-index", package, expect_error=True
```

Comment:
Note that this is failing on Python 2, because Python 2 is "bad" at encoding and we'd probably need some Python 2 specific workaround here (AHAHAHAHA). This is out of my area-of-expertise so I have approved despite this failing test. 

I personally don't care if/how we work around this -- I'm OK to skip the test on Python 2 and the NEWS file entry can say something to the effect of: "Python 2 doesn't get the new encoding validation because it's tricky there. It's a bug and we welcome PRs to fix it.". If someone wants to spend time figuring out a Python 2 specific solution for this, feel free to -- as long as it's not too complicated (which is obviously subjective), I'll be OK with it.

Reply:
Thanks! I don't think this is a bug that needs to be called out - the behavior is exactly the same as the current code (metadata only gets decoded on Python 3). So this is just not applicable to Python 2. I'll skip it.

Reply:
OK, great. Thanks for the clarification.

Rereading the comments above did seem to suggest that, but I wasn't 100% sure.

Reply:
Skipped on Python 2 [here](https://github.com/pypa/pip/pull/7539/files#diff-f13a380a0a5a8c7904168723506d1fb4R556).